moviepy/video/io/VideoFileClip.py

```diff
@@ -85,6 +85,7 @@ def __init__(self, filename, has_mask=False,
 
         # Make a reader
         pix_fmt = "rgba" if has_mask else "rgb24"
+        self.filename = None
```

Comment:
Why?

Reply:
So for the close() function is finds the reader even if we fail during its code...